### DISCUSSION TITLE
No Search in the layout

### DISCUSSION THREAD

**Discussion by PH5HP** [e107inc/e107#5274] [General]
Hi,

I have tried some themes and I found out that some themes does not show the SEARCH area in my browser although it is defined in the theme.php. On the place where this should appear there is only '1'' displayed instead. Does anyone know what is going on? Some other themes do show the SEARCH are.

Attached I have a screenshot with a piece of the ph5hp-code  of theme.php and a second screenshot with how it looks.

Kind regards,
Henk - ph5hp.nl - fileserver.ph5hp.nl

![image](https://github.com/e107inc/e107/assets/140949730/d5ffe297-4e2f-41da-bd78-62019e772ebb)
![image](https://github.com/e107inc/e107/assets/140949730/fc146409-edab-430e-8489-8c590c0c5432)

**Comment by tgtje** [comment]
@PH5HP no things to compare helas (themes maybe to old?)  However the latests versions have a search menu already 'on board' being a plugin instead of hardcoded shortcode..
So there should normally be no need for a hardcoded search (menu area) if layout has area's.
> eg menu manager column 2 > search_menu > activate and point to correct area (position of choice)..done
edit:  the used theme (BS_gaming)  uses a template for search, others might not.

**Reply by PH5HP** [reply]
Hi tgtje,

I know there is a search menu onboard and I am using the menu. I could remove the embedded one from the theme I want to use, but I like it more this way than if it is in a menu.

I did not know that BS_Gaming theme was using a template. I found it and I will investigate it's contents. Thanks for trying to help me.

UPDATE:
I still do not know why it does not work. Anyone?

Kind regards,
Henk - ph5hp.nl - fileserver.ph5hp.nl

**Comment by Vodhin** [comment]
The `{SEARCH} `  shortcode renders input elements that must be unique on the page. My guess is that the Menu Area on the right is rendering the Search shortcode before the Top area, thus the result of a second loading is "1" in the top (meaning it has already loaded).

**Comment by Alex-e107nl** [comment]
Hi, i think the shortcode is not there anymore.... i use a Jimako solution in my themes with placing this code

```
		/* {NAVBAR_SITESEARCH} */
	public function sc_sitesearch()
	{
		$text = '<form class="search-form d-flex align-items-center" method="get" action="' . e_HTTP . 'search.php">
		<input type="hidden" name="r" value="0" />
		
        <input type="text" name="q" placeholder="' . LAN_SEARCH . '" title="' . LAN_SEARCH . '" id="q" >

        <button name="s"  type="submit" title="' . LAN_SEARCH . '"><i class="bi bi-search"></i></button>
      	</form>';

		return $text;
	}
```
In the theme_shortcodes.php and use {SITESEARCH} in your theme.html to activate!

It works always :-)

**Reply by PH5HP** [reply]
Hi,

The theme has a file called 'search_template'. I guess the code for the {SEARCH} short-code will be in there, but I have no idea how this is file is linked to the theme. I zipped the file and attached it to this message.

I will also try your suggestion. TNX for that.

Kind regards,
Henk - ph5hp.nl - fileserver.ph5hp.nl

[search_template.zip](https://github.com/user-attachments/files/15809859/search_template.zip)

**Reply by PH5HP** [reply]
Hi,

I got an idea. In the source in the browser I noticed that there was something about 'search.php'. So I renamed search_template.php to search.php and that solved the problem. I have no idea why the creator named the file search_template.php instead of search.php. My only problem now is trying to get the search botton on the same line as the search input field.

Kind regards,
Henk - ph5hp.nl - fileserver.ph5hp.nl

![image](https://github.com/e107inc/e107/assets/140949730/eb2d123d-d983-493f-bcbd-71c9982c1650)

**Comment by Jimmi08** [comment]
That theme has its own repo:  
[](https://github.com/fizi/BS_Gaming_01)

From looking at it - the search template should be moved to the templates folder, nothing more.  It was changed during years.

**Comment by Jimmi08** [comment]
I fixed it for you, download it (or search_template.php only) from that repo. 

![image](https://github.com/e107inc/e107/assets/5429548/f4c32c5b-da27-468d-a37f-ab9e5cf152d8)

**Reply by PH5HP** [reply]
Hi,

Thank you. I found the repo. I works now.

Kind regards,
Henk - ph5hp.nl - fileserver.ph5hp.nl
